### PR TITLE
[JENKINS-63592] Revamp empty states

### PR DESCRIPTION
**Pull request by fqueiruga**
See [JENKINS-63592](https://issues.jenkins-ci.org/browse/JENKINS-63592).

This PR aims to revamp the empty state screens by making them more modern-looking and attractive. They style the empty state items as content blocks in order to provide a stepping stone to make them extensible through an extension point.

#### Screenshots

<details>
<summary>Anonymous user</summary>

Old
anonymous_old

New
anonymous_new
</details>

<details>
<summary>Nothing created - admin user</summary>

Old
welcome_manage_cloud_old

New
create_cloud_new2

</details>

<details>
<summary>Nothing created - user with create agent permissions</summary>

Old
welcome_create_agent_old

New
create_agent_new2

</details>

<details>
<summary>Nothing created - user with only job create permissions</summary>

Old
welcome_create_job_old

New
create_job_new2
</details>

<details>
<summary>Empty folder</summary>

Old
folder_old

New
folder_new
</details>

#### Design brief
Uploaded to the JIRA ticket

#### Breaking changes

This PR makes the previous translations obsolete. All the strings except the welcome message have been removed

### Proposed changelog entries

* Revamp the empty state screens. Their translation strings are now obsolete.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck 
@timja 
@oleg-nenashev 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by fqueiruga** [comment]
I know the git messages are bad but I wasn't able to rebase locally for some reason.

**Comment by zbynek** [comment]
Nice improvement! The "Welcome to Jenkins" title seems redundant in the empty folder -- maybe out of scope here but would be nice to remove it in that context.

**Comment by fqueiruga** [comment]
I have removed the welcome message from the empty folder screens, you were right.

The PR I think it's now ready and I have taken it out of WIP status

**Comment by timja** [comment]
the build is failing for encoding issues in the properties files

**Comment by fqueiruga** [comment]
I have fixed those encoding issues

**Comment by MRamonLeon** [comment]
Reviewed just the screenshots I see some information has been lost in the revamp, for example _to start building your software project_ Although I believe all of them may be superfluous words, I do think we should have the option to add more information to every _action_ for future options or to improve the knowledge about what the action does. I'm thinking about for example a question mark on every action which has such an extra text. Not showing the question mark if nothing is written. I say a question mark but it's more a UI design decision, what I wouldn't like is a text showing up when you pass with the mouse, to be mobile friendly. WDYT?

**Comment by fqueiruga** [comment]
I think you do have a point Ramón. Here are some options that I have sketched:

Captura de pantalla 2020-09-08 a las 11 33 31

Captura de pantalla 2020-09-08 a las 11 33 58

Btw, I appreciate ALL copyediting feedback here.

**Comment by MRamonLeon** [comment]
> Btw, I appreciate ALL copyediting feedback here.

I was thinking more about small pieces of text (a sentence or two) for every action attached to them, like the texts that were there previously.

Said that, being able to add a help link (with a different style) to every group or randomly, instead of at every single link/action, is worth as well.

**Review by timja — APPROVED** [review]

**Comment by fqueiruga** [comment]
@Ramón that would be a bit more complex I think.

@timja would you like to see any of the variants on https://github.com/jenkinsci/jenkins/pull/4926#issuecomment-688747407 implemented?

**Comment by timja** [comment]
Struggling to tell what's different. It looks good to me currently, but a before / after would help in the screenshots you added

**Comment by fqueiruga** [comment]
@timja these are the options compared

<details>
<summary>Before and after</summary>

Old
welcome_manage_cloud_old

New
welcome_manage_cloud_new
</details>

<details>
<summary>Proposed changes</summary>

Section name change:
Captura de pantalla 2020-09-08 a las 11 33 31

More info on the link:
Captura de pantalla 2020-09-08 a las 11 33 58

With link to external docs (would need a page to link to):
Captura de pantalla 2020-09-11 a las 13 26 55
</details>

Which one would you prefer? Also you @MRamonLeon , I have added your suggestion as a third option. I'm partial to the first one myself (changed section title).

**Comment by oleg-nenashev** [comment]
IMHO "Create a job" should go ahead of "Create a distributed build". Even if it is not advised, I believe the most of instances begin from running jobs on the controller

**Comment by timja** [comment]
Any thoughts on whether the states are too far to the right?

On my monitor at full screen:
image

Half size:

image

My monitor is reasonably large (27") but it does seem a bit funny.

File: core/src/main/resources/hudson/model/AllView/noJob.groovy [hunk 1]
```diff
@@ -4,40 +4,142 @@ import hudson.model.Computer
...
+                                        l.svgIcon(
+                                                class: "icon-sm",
+                                                href: "${resURL}/images/material-icons/svg-sprite-navigation-symbol.svg#ic_arrow_forward_24px")
+                                    }
+                                }
+                            }
+
+                            if (hasAdministerJenkinsPermission) {
+                                li(class: "content-block") {
+                                    a(href: "computer/new", class: "content-block__link") {
```
Comment (timja) [inline core/src/main/resources/hudson/model/AllView/noJob.groovy:42]: Hmm this doesn't seem right,

1. it's just linking to new computer, not the cloud configuration screen.
2. it's showing up when I don't have a cloud plugin installed

Although 2 probably isn't too bad as there's docs on the configure Clouds page saying what to do:

image

```suggestion
                                    a(href: "configureClouds", class: "content-block__link") {
```

Comment (fqueiruga) [reply core/src/main/resources/hudson/model/AllView/noJob.groovy:42]: Maybe I messed something up

**Comment by fqueiruga** [comment]
The content is centered within the main panel. Not sure what the best approach would be

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
I think it’s the side panel that makes it look weird on a large screen

The balance seems a bit off.

Not blocking though

**Comment by fqueiruga** [comment]
Not much we can do about that one.

**Comment by fqueiruga** [comment]
I have moved "Create a job" to the top as per Oleg's suggestion and I have also changed it's section title with copy from the former implementation.

**Review by varyvol — APPROVED** [review]

**Comment by daniel-beck** [comment]
The introductory text for

> Nothing created - user with only job create permissions

mentions distributed builds, but the action is (correctly) unavailable. This seems slightly confusing.

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Review by alecharp — APPROVED** [review]

**Pull request merged**